tools/clang/unittests/HLSLExec/CoopVec.h

```diff
@@ -354,6 +358,514 @@ GetMatrixSrcDataType(D3D12_LINEAR_ALGEBRA_DATATYPE MatrixInterpretation) {
     return D3D12_LINEAR_ALGEBRA_DATATYPE_FLOAT32;
   }
 }
+
+bool IsIntegralDataType(D3D12_LINEAR_ALGEBRA_DATATYPE DataType) {
+  return DataType == D3D12_LINEAR_ALGEBRA_DATATYPE_SINT8 ||
+         DataType == D3D12_LINEAR_ALGEBRA_DATATYPE_UINT8 ||
+         DataType == D3D12_LINEAR_ALGEBRA_DATATYPE_SINT16 ||
+         DataType == D3D12_LINEAR_ALGEBRA_DATATYPE_UINT16 ||
+         DataType == D3D12_LINEAR_ALGEBRA_DATATYPE_SINT32 ||
+         DataType == D3D12_LINEAR_ALGEBRA_DATATYPE_UINT32;
+}
+
+struct TestVector {
+private:
+  size_t NumVectors = 0;
+  size_t VectorSize = 0;
+  size_t ElementSize = 0;
+  size_t Stride = 0;
+  size_t TotalBytes = 0;
+  uint8_t *Buffer = nullptr;
+
+public:
+  TestVector(size_t NumVectors, size_t VectorSize, size_t ElementSize,
+             size_t Alignment = 16)
+      : NumVectors(NumVectors), VectorSize(VectorSize),
+        ElementSize(ElementSize) {
+    if (NumVectors == 0) {
+      throw std::invalid_argument("NumVectors must be greater than 0");
```

Comment:
nit:
nit: For new code we follow the LLVM coding standards. No brackets on single line if statements :`    if (NumVectors == 0)
      throw std::invalid_argument("NumVectors must be greater than 0");
    if (VectorSize == 0)
      throw std::invalid_argument("VectorSize must be greater than 0");
    if (ElementSize == 0)
      throw std::invalid_argument("ElementSize must be greater than 0");`